### ISSUE TITLE
Add Ouput Format Suitable For Logging

### ISSUE THREAD

**Issue by troyjohnson** [sivel/speedtest-cli#194]
I would like to request a new feature: 

An output format suitable to hand off to **logger** or put into a log file.

For example, something like:

```
ping="30.492" ping_units="ms" download="16.24" download_units="Mbit/s" upload="3.42" upload_units="Mbit/s" version="0.3.4"
```

My example may look a bit odd, but it is immediately useful in log aggregators like Splunk.

Thank you for your consideration.

**Comment by tbielawa** [comment]
I'd love to add some output enhancements as well, particularly in the case of JSON/YAML. However, unfortunately the Contributing Guidelines https://github.com/sivel/speedtest-cli/blob/master/CONTRIBUTING.md make it rather clear (strict) about exactly what can be added to this module. That's understandable though, I think the author really wants to ensure backwards compatibility all the way back to the legacy ancient archaic python versions only ever spoken of in hushed voices.

I'm working on a fork now (https://github.com/tbielawa/speedtest-cli), (no pushes, yet though) in which I intend to add alternate (machine readable) output formats as options. If I end up with a functioning implementation then I'll update this PR again w/ a reference, @troyjohnson

**Comment by sivel** [comment]
Thanks @tbielawa for the understanding.  I do intend on keeping speedtest-cli backwards compatible to python2.4 using only the standard library.

I do suppose I could be convinced to allow things such as JSON, assuming it is done in a backwards compatible way and unobtrusively.  Such as disabling those options from the option parser if required libraries are missing, or otherwise disabling them in some way, with error messages indicating the dependencies to use such functionality.

The core functionality should not be impacted by these changes however.

One of the biggest struggles I currently have is with the size that the `speedtest()` function has grown to, and it's inflexibility.  I do plan on eventual restructuring there, so if possible, please be thoughtful of complexity of changes to make it easiest to review :smile:

**Comment by eaglexboy** [comment]
@troyjohnson I currently have a fork that I modified which allows you to append results to a file in CSV format via options. I am new to python so my code will definitely reflect that. I have to make sure my code passes the contribution guidelines before I commit my changes for approval from @sivel but if you want a copy of my in-progress file just let me know.

**Comment by sivel** [comment]
The upcoming release (v1.0.0) will support CSV and JSON output, as well as the current forms of output.  This work has already been completed and can be seen on https://github.com/sivel/speedtest-cli/pull/212

**Comment by eaglexboy** [comment]
Ok, then I will not bother committing and thanks for the update.

**Comment by RERobbins** [comment]
Will the CSV output option described above allow for the possibility of generating table headers?

**Comment by sivel** [comment]
This functionality now exists in the recently released version v1.0.0
